src/app/fetch-status.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable, EventEmitter } from '@angular/core';
+import { HttpClient, HttpResponseBase } from '@angular/common/http';
+
+import { LinkData } from './link.service';
+import { BehaviorSubject } from 'rxjs';
+
+/**
+ * Internal interface to pair a status code with a boolean indicating success
+ * or failure
+ */
+interface Status {
+  code?: number;
+  ok?: boolean;
+}
+
+/**
+ * This service is responsible for checking the status of each link
```

Comment:
I think we need to extend this comment to describe interesting behavior of this class. In this case, please describe that: (i) this class check the HTTP status by sending a HEAD request, and (ii) limit the number of concurrent request.